sql/src/main/java/io/crate/metadata/FunctionInfo.java

```diff
@@ -201,17 +125,104 @@ public void readFrom(StreamInput in) throws IOException {
 
         returnType = DataTypes.fromStream(in);
         type = Type.values()[in.readVInt()];
-        deterministic = in.readBoolean();
 
-        comparisonReplacementPossible = in.readBoolean();
+        EnumSet<Feature> features = EnumSet.noneOf(Feature.class);
+        int size = in.readVInt();
+        for (int i = 0; i < size; i++) {
+            features.add(Feature.values()[in.readVInt()]);
+        }
+        // make immutable, features should never change
+        this.features = Sets.immutableEnumSet(features);
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         ident.writeTo(out);
         DataTypes.toStream(returnType, out);
         out.writeVInt(type.ordinal());
-        out.writeBoolean(deterministic);
-        out.writeBoolean(comparisonReplacementPossible);
+
+        out.writeVInt(features.size());
```

Comment:
would it make sense to encode this into a single int/long? Similar to how RegularEnumSet is implemented internally?